foyer-memory/benches/bench_dynamic_dispatch.rs

```diff
@@ -40,82 +43,77 @@ fn bench_static_dispatch<F>(t: &T<F>, loops: usize) -> Duration
 where
     F: Fn(&str) -> usize,
 {
-    let mut dur = Duration::default();
+    // Warmup
+    for _ in 0..WARMUP_LOOPS {
+        let s = rand_string(thread_rng().gen_range(0..100));
+        let _ = (t.f1)(&s);
+    }
+    let start = Instant::now();
     for _ in 0..loops {
         let s = rand_string(thread_rng().gen_range(0..100));
-        let now = Instant::now();
         let _ = (t.f1)(&s);
-        dur += now.elapsed();
     }
-    Duration::from_nanos((dur.as_nanos() as usize / loops) as _)
+    Duration::from_nanos((start.elapsed().as_nanos() as usize / loops) as _)
 }
 
 fn bench_box_dynamic_dispatch<F>(t: &T<F>, loops: usize) -> Duration
 where
     F: Fn(&str) -> usize,
 {
-    let mut dur = Duration::default();
+    // Warmup
+    for _ in 0..WARMUP_LOOPS {
+        let s = rand_string(thread_rng().gen_range(0..100));
+        let _ = (t.f3)(&s);
+    }
+    let start = Instant::now();
     for _ in 0..loops {
         let s = rand_string(thread_rng().gen_range(0..100));
-        let now = Instant::now();
         let _ = (t.f3)(&s);
-        dur += now.elapsed();
     }
-    Duration::from_nanos((dur.as_nanos() as usize / loops) as _)
+    Duration::from_nanos((start.elapsed().as_nanos() as usize / loops) as _)
 }
 
 fn bench_arc_dynamic_dispatch<F>(t: &T<F>, loops: usize) -> Duration
 where
     F: Fn(&str) -> usize,
 {
-    let mut dur = Duration::default();
+    // Warmup
+    for _ in 0..WARMUP_LOOPS {
+        let s = rand_string(thread_rng().gen_range(0..100));
+        let _ = (t.f2)(&s);
+    }
+    let start = Instant::now();
     for _ in 0..loops {
         let s = rand_string(thread_rng().gen_range(0..100));
-        let now = Instant::now();
         let _ = (t.f2)(&s);
-        dur += now.elapsed();
     }
-    Duration::from_nanos((dur.as_nanos() as usize / loops) as _)
+    Duration::from_nanos((start.elapsed().as_nanos() as usize / loops) as _)
```

Comment:
For a microbench compares the difference between direct call and a indirect call, the time of generating a random string really counts. It is not supposed to be added.

Reply:
Or, we can use `criterion` to rewrite this bench?

Reply:
Sure if you wish to help. Thanks a lot.